### PR TITLE
Add tests

### LINKED ISSUE #14: add/fix tests
currently, there are some files with name `test_` in the examples folder but they are not using the pytest fixtures.

### PR DESCRIPTION
**Pull request by ltalirz**
fix #14 

This PR adds a first complete integration test of the Isotherm workchain.

It uses the [aiida-testing](https://github.com/aiidateam/aiida-testing) package to mock the Raspa and zeo++ executables, meaning test time is spent only on the aiida-internal logic.

It also switches from Travis CI to Github actions and adds test coverage.

For review: All files in `tests/data` can be ignored.

### CODE REVIEW

File: tests/data/mock-raspa-e968334-1990dc391d097b6bec4877654b109877/_aiidasubmit.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+exec > _scheduler-stdout.txt
+exec 2> _scheduler-stderr.txt
+
+
+
+'/Users/leopold/Personal/Programmierung/qm-codes/RASPA2/src/.libs/simulate'  'simulation.input'
```
Comment (yakutovicha) [inline tests/data/mock-raspa-e968334-1990dc391d097b6bec4877654b109877/_aiidasubmit.sh]: Is this path correct?

File: tests/data/mock-raspa-e968334-3f3bf10f2df60994db3c70627a4bdd65/_aiidasubmit.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+exec > _scheduler-stdout.txt
+exec 2> _scheduler-stderr.txt
+
+
+
+'/Users/leopold/Personal/Programmierung/qm-codes/RASPA2/src/.libs/simulate'  'simulation.input'
```
Comment (yakutovicha) [inline tests/data/mock-raspa-e968334-3f3bf10f2df60994db3c70627a4bdd65/_aiidasubmit.sh]: Path?

File: tests/data/mock-raspa-e968334-6d80ddd20900011ad055b9629c6c3670/_aiidasubmit.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+exec > _scheduler-stdout.txt
+exec 2> _scheduler-stderr.txt
+
+
+
+'/Users/leopold/Personal/Programmierung/qm-codes/RASPA2/src/.libs/simulate'  'simulation.input'
```
Comment (yakutovicha) [inline tests/data/mock-raspa-e968334-6d80ddd20900011ad055b9629c6c3670/_aiidasubmit.sh]: Path?

File: tests/data/mock-zeopp-0.3-76eb9a86f7361e95317093308400b222/_aiidasubmit.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+exec > _scheduler-stdout.txt
+exec 2> _scheduler-stderr.txt
+
+
+
+'/Users/leopold/Personal/Postdoc-MARVEL/Projects/2017-09-15_pawel_plugin/zeoplusplus/network'  '-r' 'UFF.rad' '-ha' 'DEF' '-block' '1.525' '10' 'out.block' '-volpo' '1.525' '1.525' '1000' 'out.volpo' 'Mg_MOF_74.cif'
```
Comment (yakutovicha) [inline tests/data/mock-zeopp-0.3-76eb9a86f7361e95317093308400b222/_aiidasubmit.sh]: Path?

File: tests/test_parsers.py [hunk 5]
```diff
@@ -6,17 +6,14 @@
 # For further information on the license, see the LICENSE.txt file.           #
 ###############################################################################
```
Comment (yakutovicha) [inline tests/test_parsers.py]: Probably the header is wrong in this file.

**Review by yakutovicha — COMMENTED** [review]
Thanks @ltalirz 

I had a look at this PR, and, honestly, I am a bit concerned by this approach. My understanding was that the usage of mock executables was the "last resort" in case the code is not available. I believe the execution time is never a problem since it can always be shortened. Compilation can always be cached, so it is not a problem either.

Do I understand correctly that so many files are needed to perform one isotherm test? How easy would it be to maintain that, if code developers would introduce a change in the format? Would one need to replace so many files again? Especially in that particular case - where the package depends on other plugins - it can be an issue.

Please don't get me wrong - my only worry here is to avoid shooting ourselves in the foot.

**Comment by ltalirz** [comment]
>  I believe the execution time is never a problem since it can always be shortened. 

I don't think this is true in general. I can certainly imagine examples where it becomes difficult to construct tests that run very quickly - in particular if you want to make sure that your workchain works for different classes of structures and when testing workchains with many steps.
Using mock_code allows you to test the workchain using realistic use cases.

The underlying point here is that we would like to test the workchain logic and not the execution of the code binaries here.

> How easy would it be to maintain that, if code developers would introduce a change in the format?

I can see two main cases, in which you need to update the test database:

 1. If you change the inputs to one of your calculations
 2. If the plugin you are using (e.g. aiida-cp2k) drops support for the code version you were using (e.g. cp2k-7.1)

In that case, you can simply clear the test database and run the tests once locally, regenerating it from scratch.

Overall, I agree that we need to see how easy it is to maintain - this is our first time using the method, and perhaps it turns out that we need to update the test database more often than we would like to.

Given that this PR increases test coverage from 0% to 36%, I think it's worth giving it a try.

File: tests/data/mock-raspa-e968334-1990dc391d097b6bec4877654b109877/_aiidasubmit.sh (hunk 1 above)
Comment (ltalirz) [reply tests/data/mock-raspa-e968334-1990dc391d097b6bec4877654b109877/_aiidasubmit.sh]: yes, it's the path on my PC. This is the "output directory" used by the parser, i.e. the content of _aiidasubmit.sh does not matter.

Comment (yakutovicha) [reply tests/data/mock-raspa-e968334-1990dc391d097b6bec4877654b109877/_aiidasubmit.sh]: ok

**Comment by yakutovicha** [comment]
> Given that this PR increases test coverage from 0% to 36%, I think it's worth giving it a try.

Well, to me is not the main reason, there are many ways to increase the coverage :).

I don't want to block you from trying out this approach. My guess is that it will be more difficult to maintain compared to running executables as usual. Especially in this particular case. But maybe I am wrong.

**Review by yakutovicha — APPROVED** [review]

**Review by danieleongari — APPROVED** [review]
Thanks, working fine on my computer too.

**Pull request merged**